### PR TITLE
Add arguments from multiget handlers on forward relations

### PR DESCRIPTION
**Pull request by bryan-coursera**
When we have a partner -> course relation, we need to support the query parameters available on the course resource's multiget (i.e. showHidden) on the relation too. This PR refactors some things around parsing arguments and adds these arguments to all connections. It also changes the schema building behavior to fall back to the original id for related resources if the resource doesn't have a multiGet handler available.

PTAL @yifan-coursera, cc @jnwng (and @saeta because I know you still care ❤️ )

### CODE REVIEW

**Comment by saeta** [comment]
❤️

File: naptime-graphql/src/main/scala/org/coursera/naptime/ari/graphql/SangriaGraphQlSchemaBuilder.scala [hunk 1]
```diff
@@ -183,8 +143,9 @@ class SangriaGraphQlSchemaBuilder(
   def generateListHandler(
       resource: Resource,
       handler: Handler): Field[SangriaGraphQlContext, DataMap] = {
-    val arguments = generateHandlerArguments(handler)
     val resourceName = ResourceName(resource.name, resource.version.getOrElse(0L).toInt)
+    val arguments = SangriaGraphQlSchemaBuilder.generateHandlerArguments(handler)
+
```
Comment (yifan-coursera) [inline naptime-graphql/src/main/scala/org/coursera/naptime/ari/graphql/SangriaGraphQlSchemaBuilder.scala]: Remove stray newline.

File: naptime-graphql/src/main/scala/org/coursera/naptime/ari/graphql/SangriaGraphQlSchemaBuilder.scala [hunk 2]
```diff
@@ -197,25 +158,18 @@ class SangriaGraphQlSchemaBuilder(
...
-        val tpe = parameter.`type`
-        // TODO(bryan): Use argument defaults here
-        Argument(
-          name = parameter.name,
-          argumentType = scalaTypeToSangria(tpe))(scalaTypeToFromInput(tpe), implicitly).asInstanceOf[Argument[Any]]
-      }.toList
+    val mergedArguments = (field.arguments ++ arguments)
+      .groupBy(_.name)
+      .map(_._2.head)
+      .map(_.asInstanceOf[Argument[Any]])
```
Comment (yifan-coursera) [inline naptime-graphql/src/main/scala/org/coursera/naptime/ari/graphql/SangriaGraphQlSchemaBuilder.scala]: Why do we need to strip the types?

File: naptime-graphql/src/main/scala/org/coursera/naptime/ari/graphql/schema/NaptimePaginatedResourceField.scala [hunk 3]
```diff
@@ -19,18 +21,29 @@ object NaptimePaginatedResourceField {
...
-        val limit = args.arg(NaptimePaginationField.limitArgument)
-        Math.max(limit / 10, 1) * COMPLEXITY_COST * childScore
-      }),
-      arguments = NaptimePaginationField.paginationArguments)
+      fieldName: String): Option[Field[SangriaGraphQlContext, DataMap]] = {
+
+    val resource = schemaMetadata.getResource(resourceName)
+
+    for {
+      schema <- schemaMetadata.getSchema(resource)
```
Comment (yifan-coursera) [inline naptime-graphql/src/main/scala/org/coursera/naptime/ari/graphql/schema/NaptimePaginatedResourceField.scala]: Is `schema` used anywhere?

**Comment by bryan-coursera** [comment]
Merged in with #147

**Pull request closed without merging**